RockLib.Configuration.Conditional/ConditionalConfigurationSection.cs

```diff
@@ -0,0 +1,118 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace RockLib.Configuration.Conditional;
+
+/// <summary>
+/// A ConfigurationSection that can use an alternate ConfigurationSection to
+/// override some values.
+/// </summary>
+public class ConditionalConfigurationSection : IConfigurationSection
+{
+    private readonly IConfigurationSection _baseSection;
+    private readonly Func<IConfigurationSection> _getOverrideSection;
+
+    private ConditionalConfigurationSection(IConfigurationSection baseSection, Func<IConfigurationSection> getOverrideSection)
+    {
+        _baseSection = baseSection;
+        _getOverrideSection = getOverrideSection;
+    }
+
+    /// <summary>
+    /// Create a ConditionalConfigurationSection instance with overrides from a
+    /// child section determined by a switching property.
+    /// </summary>
+    /// <param name="baseSection">
+    ///   The base IConfigurationSection to use for this configuration
+    /// </param>
+    /// <param name="switchingProperty">
```

Comment:
Should we rename this to something more in line with the new conditional name?  conditionalProperty?